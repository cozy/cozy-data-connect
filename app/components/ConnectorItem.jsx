/** @jsx h */
import { h } from 'preact'
import { Link, withRouter } from 'react-router'
import { translate } from '../plugins/preact-polyglot'

const ConnectorItem = ({ title, subtitle, connected, slug, iconName, backgroundCSS = 'white', enableDefaultIcon = false, router }) => (
  <Link class='item-wrapper' to={`${router.location.pathname}/${slug}`}>
    {backgroundCSS &&
    <header class='item-header' style={{background: backgroundCSS}}>
      {iconName &&
        <svg class='item-icon'>
          <use xlinkHref={icon(iconName, enableDefaultIcon)} />
        </svg>
      }
    </header>
    }
    {!backgroundCSS &&
    <header class='item-header'>
      {iconName &&
        <svg class='item-icon'>
          <use xlinkHref={icon(iconName, enableDefaultIcon)} />
        </svg>
      }
    </header>
    }
    <p class='item-title'>{title}</p>
    {subtitle && <p class='item-subtitle'>{subtitle}</p>}
    {connected &&
      <svg class='item-connected'>
        <use xlinkHref={require('../assets/sprites/icon-check.svg')} />
      </svg>
    }
  </Link>
)

// Fallback to get the item icon and avoid error if not found
// with a possible default icon
const icon = (iconName, enableDefaultIcon) => {
  let icon = ''
  try {
    icon = require(`../assets/icons/${iconName}.svg`)
  } catch (e) {
    if (enableDefaultIcon) {
      icon = require('../assets/icons/default_myaccount.svg')
    }
  }
  return icon
}

export default translate()(withRouter(ConnectorItem))
