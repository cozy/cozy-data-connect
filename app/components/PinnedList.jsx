/** @jsx h */
import { h } from 'preact'
import { translate } from '../plugins/preact-polyglot'
import ConnectorList from './ConnectorList'

const PinnedList = ({ t, connectors, children }) => (
  <div class='content'>
    <h1>{t('my_accounts pinned title')}</h1>
    <ConnectorList connectors={connectors} />
    {children}
  </div>
)

export default translate()(PinnedList)
