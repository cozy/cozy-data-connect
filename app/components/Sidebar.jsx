/** @jsx h */
import { h } from 'preact'
import { Link, withRouter } from 'react-router'
import { translate } from '../plugins/preact-polyglot'

const Sidebar = ({ t, categories, router }) => {
  let isCategoryView = router.location.pathname.match(/^\/category/) !== null
  const i18nCategories =
    categories.filter(c => c !== 'others')
      .map(c => ({ slug: c, label: t(`${c} category`) }))
      .sort((a, b) => a.label.localeCompare(b.label))

  return (
    <aside className='coz-sidebar'>
      <ul role='navigation' className='coz-nav'>
        <li className='coz-nav-item'>
          <Link to='/discovery' className='coz-nav-link con-cat-discovery' activeClassName='active'>
            {t('my_accounts discovery title')}
          </Link>
        </li>
        <li className='coz-nav-item'>
          <Link to='/category/all'
            className={isCategoryView ? 'coz-nav-link con-cat-categories active' : 'coz-nav-link con-cat-categories'}>
            {t('my_accounts category title')}
          </Link>
          {isCategoryView &&
            <ul className='con-nav-submenu'>
              <li>
                <Link to='/category/all' className='con-nav-submenu-link' activeClassName='active'>
                  {t('all category')}
                </Link>
              </li>
              {i18nCategories.map(category => (
                <li>
                  <Link to={`/category/${category.slug}`} className='con-nav-submenu-link' activeClassName='active'>
                    {category.label}
                  </Link>
                </li>
              ))}
              <li>
                <Link to='/category/others' className='con-nav-submenu-link' activeClassName='active'>
                  {t('others category')}
                </Link>
              </li>
            </ul>
          }
        </li>
        <li className='coz-nav-item'>
          <Link to='/connected' className='coz-nav-link con-cat-pinned' activeClassName='active'>
            {t('my_accounts connected title')}
          </Link>
        </li>
      </ul>
    </aside>
  )
}
export default translate()(withRouter(Sidebar))
