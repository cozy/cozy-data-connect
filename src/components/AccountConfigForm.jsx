import React from 'react'

import Field, { DropdownField } from './Field'
import AccountLoginForm from './AccountLoginForm'

// Ensure that path has a / as first character
const sanitizePath = path => path.charAt(0) === '/' ? path : `/${path}`

const folderLink = (path, folders) => {
  path = sanitizePath(path)
  let folder = folders.find(f => f.path + '/' + f.name === path)
  if (!folder) return
  return `/apps/files/#folders/${folder.id}`
}

const AccountConfigForm = ({ t, customView, fields, dirty, error, submit, submitting }) => (
  <div className={'account-form' + (error ? ' error' : '')}>
    {fields.folderPath &&
      <div>
        <h3>{t('location')}</h3>
        <p>{t('location desc')}</p>
        <DropdownField label={false} {...fields.folderPath} />
        <a href={folderLink(fields.folderPath.value, fields.folderPath.folders)} target='_blank'>
          {t('location button')}
        </a>
      </div>
    }
    {fields.calendar &&
      <div>
        <h3>{t('calendar')}</h3>
        <p>{t('calendar desc')}</p>
        <Field label={false} {...fields.calendar} />
      </div>
    }
    <h3>{t('frequency')}</h3>
    <p>{t('frequency desc')}</p>
    <Field label={false} {...fields.frequency}>
      <select className='account-field-dropdown'>
        <option value='none'>{t('none')}</option>
        <option value='hour'>{t('every hour')}</option>
        <option value='day'>{t('every day')}</option>
        <option value='week'>{t('every week')}</option>
        <option value='month'>{t('each month')}</option>
      </select>
    </Field>
    <h3>{t('account')}</h3>
    <AccountLoginForm
      t={t}
      customView={customView}
      fields={fields}
    />
    {error === 'bad credentials' &&
      <p className='errors'>{t('account config bad credentials')}</p>
    }
  </div>
)

export default AccountConfigForm
