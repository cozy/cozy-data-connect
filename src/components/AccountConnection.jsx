import React, { Component } from 'react'
import statefulForm from '../lib/statefulForm'

import AccountLoginForm from './AccountLoginForm'
import DataItem from './DataItem'
import ReactMarkdown from 'react-markdown'

class AccountConnection extends Component {
  constructor (props, context) {
    super(props, context)

    this.state = {
      submitting: false
    }

    this.submit = () => {
      this.setState({
        submitting: true
      })

      this.props.submit()
        .catch(error => this.setState({submitting: false, error: error.message}))
    }
  }

  render () {
    const { t, connector, fields, error } = this.props
    let { dirty } = this.props
    const { name, description } = connector
    const { submitting } = this.state

    let button = <button disabled={!dirty} aria-busy={submitting ? 'true' : 'false'}
      onClick={() => this.submit()}>
      {t('account config button')}
    </button>

    if (connector.oauth) {
      // TODO find properly the root cozy url (without data-account)
      button = <a target='_blank' href='http://cozy.tools:8080/accounts/orange/start'>Submit</a>
    }

    // If there is no field displayed, the form is dirty by default.
    dirty = dirty || Object.values(fields).every(field => field.type === 'hidden' || field.advanced)
    return (
      <div className='account-connection'>
        <div className='account-description'>
          <h3>{t('title description')}</h3>
          <p>
            <ReactMarkdown
              source={
                t(description)
              }
              renderers={{Link: props => <a href={props.href} target='_blank'>{props.children}</a>}}
            />
          </p>
          <h3>{t('dataType title')}</h3>
          <ul className='account-datas'>
            {connector.dataType && connector.dataType.map(data =>
              <DataItem
                dataType={data}
                hex={connector.color.hex}
              />
            )}
          </ul>
          <p>{` ${connector.name} ${t('dataType disclaimer')} `}</p>
        </div>
        <div className='account-login'>
          <h3>{t('account config title', {name: name})}</h3>
          <div className={'account-form' + (error ? ' error' : '')}>
            <AccountLoginForm
              t={t}
              fields={fields}
            />
            <div className='account-form-controls'>
              {button}
              {error === 'bad credentials' &&
                <p className='errors'>{t('account config bad credentials')}</p>
              }
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default statefulForm()(AccountConnection)
