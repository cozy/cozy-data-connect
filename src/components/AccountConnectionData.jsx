import styles from '../styles/accountConnectionData'

import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import ClassNames from 'classnames'
import ReactMarkdownWrapper from '../components/ReactMarkdownWrapper'
import DataItem from '../components/DataItem'

import { getKonnectorMessage } from '../lib/konnectors'

const AccountConnectionData = ({ t, connector }) => {
  const hasDataTypes = !!(connector.data_types && connector.data_types.length)
  const serviceMessage = getKonnectorMessage(t, connector, 'service')
  return (
    <div className={ClassNames(styles['col-account-connection-data'])}>
      {serviceMessage && (
        <div>
          <h4>{t('account.config.data.service.description')}</h4>
          <p>
            <ReactMarkdownWrapper source={serviceMessage} />
          </p>
        </div>
      )}
      <h4>{t('account.config.data.title')}</h4>
      {hasDataTypes && (
        <ul className={styles['col-account-connection-data-access']}>
          {connector.data_types.map(data => <DataItem dataType={data} />)}
        </ul>
      )}
      {!hasDataTypes && <p>{t('dataType.none', { name: connector.name })}</p>}
    </div>
  )
}

export default translate()(AccountConnectionData)
