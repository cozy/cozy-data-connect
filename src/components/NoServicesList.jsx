import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { Title, Text } from 'cozy-ui/react/Text'
import { Media, Img, Bd } from 'cozy-ui/transpiled/react/Media'

import CandidateServiceTile from 'components/CandidateServiceTile'
import CandidateCategoryTile from 'components/CandidateCategoryTile'
import AddServiceTile from 'components/AddServiceTile'
import candidatesConfig from 'config/candidates'
import ArrowIllustration from 'assets/images/drawing-arrow-up.svg'

export const NoServicesList = ({ t }) => (
  <>
    <div className="services-list">
      {candidatesConfig.konnectors.map(({ slug, name }) => (
        <CandidateServiceTile key={slug} slug={slug} name={name} />
      ))}
      {Object.entries(candidatesConfig.categories).map(([category, slugs]) => (
        <CandidateCategoryTile
          key={category}
          slugs={slugs}
          category={category}
        />
      ))}
      <AddServiceTile label={t('add_service')} />
    </div>
    <Media align="top" className="EmptyServicesListTip">
      <Img>
        <img src={ArrowIllustration} />
      </Img>
      <Bd className="EmptyServicesListTip-text">
        <Title>{t('connector.empty.title')}</Title>
        <Text tag="p">{t('connector.empty.text')}</Text>
      </Bd>
    </Media>
  </>
)

export default translate()(NoServicesList)
