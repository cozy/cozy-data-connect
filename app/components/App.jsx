/** @jsx h */
import { h } from 'preact'

import styles from '../styles/app.styl'
import classNames from 'classnames'

import Sidebar from './Sidebar'
import Notifier from './Notifier'

const App = ({ categories, children }) => (
  <div className={classNames(styles['con-wrapper'], styles['coz-sticky'])}>
    <Sidebar categories={categories} />
    <main className={styles['con-content']}>
      {children}
    </main>
    <Notifier />
  </div>
)

export default App
