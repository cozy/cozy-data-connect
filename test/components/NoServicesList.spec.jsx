'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import { NoServicesList } from '../../src/components/NoServicesList'
import { tMock } from '../jestLib/I18n'

describe('NoServicesList component', () => {
  it('should render service suggestions', () => {
    const appSuggestions = { data: [{ slug: 'test1' }, { slug: 'test2' }] }
    const component = shallow(
      <NoServicesList t={tMock} appSuggestions={appSuggestions} />
    )
    expect(component.getElement()).toMatchSnapshot()
  })
})
