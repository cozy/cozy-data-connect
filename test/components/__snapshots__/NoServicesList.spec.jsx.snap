// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoServicesList component should render service suggestions 1`] = `
<React.Fragment>
  <div
    className="services-list"
  >
    <Wrapper
      name="Ameli"
      slug="ameli"
    />
    <Wrapper
      name="Impôts.gouv.fr"
      slug="impots"
    />
    <Wrapper
      category="banking"
      slugs={
        Array [
          "hsbc119",
          "caissedepargne1",
          "ing",
          "cic45",
        ]
      }
    />
    <Wrapper
      category="isp"
      slugs={
        Array [
          "sfr",
          "free",
          "orange",
          "bouygues",
        ]
      }
    />
    <Wrapper
      category="insurance"
      slugs={
        Array [
          "maifecheancier",
          "alan",
          "harmonie",
          "macif",
        ]
      }
    />
    <Wrapper
      category="energy"
      slugs={
        Array [
          "engie",
          "planeteoui",
          "veoliaeau",
          "ekwateur",
        ]
      }
    />
    <withClient(AddServiceTile)
      label="Add"
    />
  </div>
  <Media
    align="top"
    className="EmptyServicesListTip"
  >
    <Img>
      <img
        src="test-file-stub"
      />
    </Img>
    <Bd
      className="EmptyServicesListTip-text"
    >
      <Title
        ellipsis={false}
        tag="div"
      >
        Start gathering your data!
      </Title>
      <Text
        ellipsis={false}
        tag="p"
      >
        Synchronise your brands with your Cozy to automatically retrieve your data (bills, reimbursements, expenses…)
      </Text>
    </Bd>
  </Media>
</React.Fragment>
`;
